platform/ios/platform/ios/src/MGLScaleBar.mm

```diff
@@ -282,8 +301,44 @@ - (MGLRow)preferredRow {
     return *table;
 }
 
+
+#pragma mark - Dark Mode Changes
+
+- (void)setShouldShowDarkStyles:(BOOL)shouldShowDarkStyles {
+    
+    if (_shouldShowDarkStyles != shouldShowDarkStyles) {
+        _shouldShowDarkStyles = shouldShowDarkStyles;
+        
+        // Redraw labels
+        _prototypeLabel.shouldShowDarkStyles = shouldShowDarkStyles;
+        [self resetLabelImageCache];
+        [self updateLabels];
+        
+    }
+}
+
+
 #pragma mark - Setters
 
+- (void)setUsesMetricSystem:(BOOL)usesMetricSystem {
+    
+    if (_usesMetricSystem != usesMetricSystem) {
+        
+        _usesMetricSystem = usesMetricSystem;
+        
+        // Set the distance formatter locale using Germany for metric and United States for imperial.
+        self.formatter.numberFormatter.locale = usesMetricSystem ? [NSLocale localeWithLocaleIdentifier:@"de_DE"] : [NSLocale localeWithLocaleIdentifier:@"en_US"];
```

Comment:
This produces the case of `1,5 km`, but is there locale, say in en-US, where I want metric, but a `.` for a decimal point, such as `1.5 km`?

Reply:
You made a good point about this. I somehow didn't catch this. Will put in a PR to fix this.